### PR TITLE
Issue/877 final improvements

### PR DESCRIPTION
**Pull request by savacano28**
### Proposed changes
* Hoover with launch time detail, icon clickable to open updateInject and save checkbox state in localstorage ignoring ids.
* Refact ButtonPopover component, injectIcon
* Fix render error in dateScheduleExercises
* Fix queries scenarios/exercise: team/exercise/inject(x) 

Notion with more details : https://www.notion.so/filigran/Visualize-the-timeline-of-your-injects-in-Scenario-70c2e4a31eef475aaf047a35cebdc406

### Related issues

* #877
*

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: openbas-front/src/components/Timeline.tsx [hunk 1]
```diff
@@ -89,9 +89,10 @@ interface Props {
   exerciseOrScenarioId: string,
   injects: Inject[],
```
Comment (savacano28) [inline openbas-front/src/components/Timeline.tsx:90]: @RomuDeuxfois

File: openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts [hunk 2]
```diff
@@ -11,8 +12,9 @@ const injectContextForScenario = (scenario: ScenarioStore) => {
     onAddInject(inject: Inject): Promise<{ result: string }> {
       return dispatch(addInjectForScenario(scenario.scenario_id, inject));
     },
-    onUpdateInject(injectId: Inject['inject_id'], inject: Inject): Promise<{ result: string }> {
-      return dispatch(updateInjectForScenario(scenario.scenario_id, injectId, inject));
+    async onUpdateInject(injectId: Inject['inject_id'], inject: Inject): Promise<{ result: string }> {
+      await dispatch(updateInjectForScenario(scenario.scenario_id, injectId, inject));
+      return dispatch(fetchScenarioTeams(scenario.scenario_id));
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts:17]: Why we need to refetch scenario teams here ?
Is not really performant

File: openbas-front/src/components/common/ButtonPopover.tsx [hunk 3]
```diff
@@ -39,11 +44,25 @@ const ButtonPopover: FunctionComponent<Props> = ({
...
         {entries.map((entry, idx) => {
+          if (entry.type === 'export') {
+            return (
+              <MenuItem key={idx}>
+                <CSVLink
+                  data={entry.data || []}
+                  filename={entry.filename || 'data.xls'}
+                >
+                  {t(entry.label)}
+                </CSVLink>
```
Comment (RomuDeuxfois) [inline openbas-front/src/components/common/ButtonPopover.tsx:55]: It's not supposed to be here
This component is generic and you are putting a lot of specific stuff inside.

**Review by RomuDeuxfois — CHANGES_REQUESTED** [review]

File: openbas-api/src/main/java/io/openbas/rest/helper/TeamHelper.java [hunk 4]
```diff
@@ -109,11 +111,33 @@ public static List<TeamSimple> rawTeamToSimplerTeam(List<RawTeam> teams,
...
             // We set the injects linked to the scenarios
-            teamSimple.setScenariosInjects(rawTeam.getTeam_scenarios().stream().flatMap(
+            teamSimple.setScenariosInjects(
+                getInjectTeamsIds(teamSimple.getId(),
+                rawTeam.getTeam_scenarios().stream().flatMap(
                     scenario -> mapInjectsByScenarioIds.get(scenario).stream()
-            ).collect(Collectors.toSet()));
+                ).collect(Collectors.toSet()),
+                injectRepository)
+            );
```
[the remark this answers is not in the document]
Comment (savacano28) [reply openbas-api/src/main/java/io/openbas/rest/helper/TeamHelper.java:120]: Sure! that works too. I'll update the code.

File: openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts (hunk 2 above)
Comment (savacano28) [reply openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts:17]: Sure! I agree with you, but we need to get the teams updated here. We have to identify technical teams, injects without teams, and even teams without injects. For that, I re-fetch the teams. However, I could try passing just the injects and see what we can show.

Comment (RomuDeuxfois) [reply openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts:17]: The injects contains the new team ids but it's maybe not straightforward to re map to technical teams, injects without teams, and even teams without injects.
This is what you mean ?

File: openbas-api/src/main/java/io/openbas/rest/helper/TeamHelper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/helper/TeamHelper.java:120]: And you probably don't need a new onject RawInjectTeam, just a RawInject

File: openbas-front/src/admin/components/common/injects/Injects.js [hunk 5]
```diff
@@ -175,7 +175,12 @@ const Injects = (props) => {
   const classes = useStyles();
   const { t, tPick } = useFormatter();
   const [selectedInjectId, setSelectedInjectId] = useState(null);
-  const [showTimeline, _setShowTimeline] = useState(true);
+  const [showTimeline, setShowTimeline] = useState(
+    () => {
+      const storedValue = localStorage.getItem(exerciseOrScenarioId);
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/common/injects/Injects.js:180]: we need a more specific key to understand what is store in the localstorage
${exerciseOrScenarioId}_injects_timeline or something else

File: openbas-front/src/admin/components/simulations/simulation/ExerciseDatePopover.tsx [hunk 6]
```diff
@@ -24,9 +24,11 @@ const ExerciseDatePopover: React.FC<Props> = ({ exercise }) => {
   return (
     <>
       <Tooltip title={(t('Modify the scheduling'))}>
-        <IconButton size="small" color="primary" onClick={() => setOpenEdit(true)} style={{ marginRight: 5 }} disabled={exercise.exercise_status !== 'SCHEDULED'}>
-          <UpdateOutlined />
-        </IconButton>
+        <span>
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/simulations/simulation/ExerciseDatePopover.tsx:27]: Why did you add this span here?

File: openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts (hunk 2 above)
Comment (savacano28) [reply openbas-front/src/admin/components/scenarios/scenario/ScenarioContext.ts]: Fixed with Romuald modifs
